pkg/ottl/ottlfuncs/func_decode.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"golang.org/x/text/encoding/ianaindex"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type DecodeArguments[K any] struct {
+	Target   ottl.Getter[K]
+	Encoding string
+}
+
+func NewDecodeFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Decode", &DecodeArguments[K]{}, createDecodeFunction[K])
+}
+
+func createDecodeFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*DecodeArguments[K])
+	if !ok {
+		return nil, fmt.Errorf("DecodeFactory args must be of type *DecodeArguments[K]")
+	}
+
+	return Decode(args.Target, args.Encoding)
+}
+
+func Decode[K any](target ottl.Getter[K], encoding string) (ottl.ExprFunc[K], error) {
```

Comment:
Is there logic we can reuse from pkg/stanza?

Reply:
just had a look at stanza and while base64 is not supported there we might be able to use some logic for the other encodings. will look into that now and adapt the PR if we can use something from there

Reply:
Alright, some of the logic could be reused from stanza. just pushed the changes

Reply:
I'm okay with the current overall approach, but could we duplicate or share the encoding overrides that Stanza uses? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/decode/decode.go#L49

Changing these later would be a breaking change, so I figure we should, at a minimum, duplicate them for now.

Reply:
@bacherfl has this been addressed yet?

Reply:
@TylerHelmuth sorry seems like I overlooked that one - I have added the encodingOverrides from stanza now to be considered here as well